unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs

```diff
@@ -324,9 +341,8 @@ internal void UpdateAvailableRealmsInfo(IEnumerable<RealmModel> currentRealmList
                 {
                     name = realmModel.serverName,
                     players = realmModel.usersCount,
-                    isConnected = realmModel.serverName == currentRealm?.serverName
+                    isConnected = realmModel.serverName == serverName
                 };
-
                 currentAvailableRealms.Add(realmToAdd);
```

Comment:
Can you also try if converting main parameter to `List` doesn't brake anything? Because Rider suggest it 
```
internal void UpdateAvailableRealmsInfo(List<RealmModel> currentRealmList)
    {
        if (!NeedToRefreshRealms(currentRealmList))
            return;

        currentAvailableRealms.Clear();
        
        if (currentRealmList != null)
        {
            string serverName = ServerNameForCuurentRealm();
            
            foreach (RealmModel realmModel in currentRealmList)
                currentAvailableRealms.Add(new RealmRowComponentModel
                {
                    name = realmModel.serverName,
                    players = realmModel.usersCount,
                    isConnected = realmModel.serverName == serverName
                });
        }

        view.currentRealmSelectorModal.SetAvailableRealms(currentAvailableRealms);
    }
    
    private static string ServerNameForCuurentRealm()
    {
        if (DataStore.i.realm.playerRealm.Get() != null)
            return DataStore.i.realm.playerRealm.Get().serverName;

        if (DataStore.i.realm.playerRealmAbout.Get() != null)
            return DataStore.i.realm.playerRealmAbout.Get().Configurations.RealmName;

        return string.Empty;
    }
```

Reply:
because otherwise Rider informs about possible multiple enumeration and object allocation (and making function argument to be `List` seems to remove it)
![image](https://user-images.githubusercontent.com/35366872/199252947-08057cef-d90c-4901-b1ec-aa9a94d07007.png)

![image](https://user-images.githubusercontent.com/35366872/199252848-9bc3424a-99dd-4237-8a60-db3904e5d71a.png)